src/main/java/uk/nhs/tis/sync/job/CommonSyncJobTemplate.java

```diff
@@ -31,11 +31,16 @@ public abstract class CommonSyncJobTemplate<T> implements RunnableJob {
 
   protected Stopwatch mainStopWatch;
 
-  @Autowired
-  private EntityManagerFactory entityManagerFactory;
+  protected final ApplicationEventPublisher applicationEventPublisher;
+
+  private final EntityManagerFactory entityManagerFactory;
 
-  @Autowired(required = false)
-  protected ApplicationEventPublisher applicationEventPublisher;
+  @Autowired
```

Comment:
I ran the app to check.  As it's the only constructor, we don't need to annotate them.